com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs

```diff
@@ -876,6 +955,37 @@ protected void AddRenderPasses(ref RenderingData renderingData)
                 m_UseOptimizedStoreActions = false;
         }
 
+        /// <summary>
+        /// Calls <c>Setup</c> for each feature added to this renderer.
+        /// <seealso cref="ScriptableRendererFeature.SetupRenderPasses(ScriptableRenderer, in RenderingData)"/>
+        /// </summary>
+        /// <param name="renderingData"></param>
+        protected void SetupRenderPasses(in RenderingData renderingData)
```

Comment:
If it is in `AddRenderPasses`, some of the targets might not have been allocated yet.
This was fine when using RTI because there was no assignment when running `GetTemporaryRT` and it could be done after that identifier was referred to in a render feature.
The problem with RTHandles is that they are null before allocation. Some renderer features might try to get this reference too early in `AddRenderPasses` and run into trouble. `SetupRenderPasses` is called from Universal Renderer after the targets are created and the RTHandles replace the nulls.

See [`DecalRendererFeature.cs`](https://github.com/Unity-Technologies/Graphics/pull/5346/files#diff-ae4b757f9fbac3814946eb9c7c1a285ea516ac85ec9b2532329f47326dfa3ac5)

Reply:
Ahh okay, that makes sense. Does this have any impact on user render passes?